addons/radar/functions/fnc_compassPFH.sqf

```diff
@@ -40,7 +41,8 @@ if ([_player] call EFUNC(main,canHudBeShown)) then {
         {diwako_dui_enable_compass_dir isEqualTo 1 && {!(isNull objectParent _player)} ||
         {diwako_dui_enable_compass_dir isEqualTo 2} ||
         {diwako_dui_enable_compass_dir isEqualTo 3 && {private _veh = (vehicle _player); _veh isNotEqualTo _player && {(driver _veh) isEqualTo _player}}}
-        }}) then {
+        }} ||
+        _hasGPS && {diwako_dui_enable_compass_dir isEqualTo 4}) then {
```

Comment:
I admit, that part is a curly bracket hell, I would suggest doing it like so

```sqf
    if (_hasCompass &&
        {diwako_dui_enable_compass_dir in [1, 2, 3, 4] &&
        {diwako_dui_enable_compass_dir isEqualTo 1 && {!(isNull objectParent _player)} ||
        {diwako_dui_enable_compass_dir isEqualTo 2} ||
        {diwako_dui_enable_compass_dir isEqualTo 3 && {private _veh = (vehicle _player); _veh isNotEqualTo _player && {(driver _veh) isEqualTo _player}}} || 
        {diwako_dui_enable_compass_dir isEqualTo 4 && {([_player] call FUNC(getGPS)) isNotEqualTo ""}}
        }}) then {     
```

With this the `_hasGPS` variable does not need to be written all the time, the `_hasCompass` variable is used later again hence why it is written there.

Reply:
> I admit, that part is a curly bracket hell

Yeah, it took me a minute to figure out that part. :D

Originally, I wanted to separate `_hasGPS` from `_hasCompass` so the player wouldn't be required to carry both when the GPS check is enabled.
I suppose this could work as well, also it looks cleaner.